File: IperfManager/IperfServer.py
import PyQt5

from AbstractIperf import AbstractIperf
from typing import List
# from PySide6.QtCore import QObject, Signal, Slot, QProcess
from PyQt5.QtCore import QProcess, QObject, pyqtSignal, pyqtSlot


class IperfServer(AbstractIperf, QObject):
    server_started = pyqtSignal(bool)

    def __init__(self, program: str, args: list[str], parent=None):
        super().__init__()
        self.process = QProcess()
        self.process.started.connect(self.process_started)
        self.process.finished.connect(self.process_finished)
        self.process.readyRead.connect(self.process_output)

        self.program = program
        self.args = args

        self.parent = parent

    def start(self):
        self.process.start(self.program, self.args)

    @pyqtSlot()
    def process_output(self):
        output = self.process.readAll()
        print(output)

    @pyqtSlot()
    def process_started(self):
        print('IperfServer started')
        self.server_started.emit(True)

    @pyqtSlot()
    def process_finished(self):
        print('IperfServer finished')

File: IperfManager/src/iperfserver.cpp
#include "iperfserver.h"

IperfServer::IperfServer()
{
    process = new QProcess();
    connect(process, &QProcess::started, this, &IperfServer::started);
    connect(process, &QProcess::readyReadStandardOutput, this, &IperfServer::output);

//    connect(process, QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished),
//        [=](int exitCode, QProcess::ExitStatus exitStatus){

//    });

}

IperfServer::~IperfServer()
{
    process->deleteLater();
}

void IperfServer::setParams(const QString &program, const QStringList &args)
{
    process->setProgram(program);
    process->setArguments(args);
}

void IperfServer::start()
{
    process->start();
    process->waitForReadyRead();
}

void IperfServer::stop()
{

}

void IperfServer::waitForFinished(int sec)
{
    process->waitForFinished(sec);
}

void IperfServer::output()
{
    qDebug() << process->readAll();
}

bool IperfServer::started()
{
    qDebug() << "Процесс запустился";
    return true;
}
